bundles/framework/announcements/view/AnnouncementsHandler.js

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import { StateHandler, controllerMixin, Messaging } from 'oskari-ui/util';
+import { Message } from 'oskari-ui';
+
+// Handler for announcements. Handles state and service calls.
+
+const getMessage = (key, args) => <Message messageKey={key} messageArgs={args} bundleKey='announcements' />;
+
+class ViewHandler extends StateHandler {
+    constructor () {
+        super();
+        this.state = {
+            panels: [],
+            modals: [],
+            updated: false,
+            announcements: [],
+            checked: false
+        };
+    }
+
+    getAnnouncements (callback) {
+        if (typeof callback !== 'function') {
+            return;
+        }
+
+        jQuery.ajax({
+            type: 'GET',
+            dataType: 'json',
+            url: Oskari.urls.getRoute('Announcements'),
+            success: function (pResp) {
+                callback(pResp);
+            },
+            error: function (jqXHR, textStatus) {
+                Messaging.error(getMessage('messages.getFailed'));
+            }
+        });
+    }
```

Comment:
Trigger loading on startup, not on the component at render time:
```suggestion
        };
        this.fetchAnnouncements();
    }

    fetchAnnouncements () {
        jQuery.ajax({
            type: 'GET',
            dataType: 'json',
            url: Oskari.urls.getRoute('Announcements'),
            success: (pResp) => {
                this.updateState({
                    announcements: pResp.data
                });
            },
            error: function (jqXHR, textStatus) {
                Messaging.error(getMessage('messages.getFailed'));
            }
        });
    }
```